examples/plot_incr_kmeans.py

```diff
@@ -0,0 +1,319 @@
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from cardinal.kmeans import IncrementalMiniBatchKMeans
+from cardinal.clustering import MiniBatchKMeansSampler, IncrementalMiniBatchKMeansSampler
+from sklearn.datasets import make_blobs
+from matplotlib import pyplot as plt
+from scipy.stats import ttest_rel
+from cardinal.utils import ActiveLearningSplitter
+import numpy as np
+
+# Inertia
+# ^^^^^^^
+#
+# We will be talking a  lot about inertia in this notebook which is the
+# euclidean distance from the points in a dataset to their closest center.
+# We define a function to measure it.
+
+
+def inertia(data, centers):
+    return (pairwise_distances_argmin_min(data, centers)[1] ** 2).sum()
+
+
+
+# Active learning like-experiment
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# One core element of the experiment differs from active learning. In fact,
+# in active learning, we do not expect to select a number of samples equal to
+# the number of real cluster in the data. We also build our clustering over
+# several iterations. We now try to do the same on generated data.
+
+X, y, centers = make_blobs(n_samples=10000, centers=100, return_centers=True,
+                           n_features=512, random_state=2)
+
+clf = RandomForestClassifier()
+
+kmeans_inertia = []
+kmeans_accuracy = []
+sampler = MiniBatchKMeansSampler(10)
+
+idx = ActiveLearningSplitter(10000, test_size=.2, random_state=0)
+idx.add_batch(np.arange(10))
+
+for i in range(10):
+    selected = sampler.fit(X[idx.selected]).select_samples(X[idx.non_selected])
+    idx.add_batch(selected)
+    kmeans_inertia.append(inertia(X[idx.test], X[idx.selected]))
+
+    clf.fit(X[idx.selected], y[idx.selected])
+    kmeans_accuracy.append(clf.score(X[idx.test], y[idx.test]))
+    
+
+ikmeans_inertia = []
+ikmeans_accuracy = []
+sampler = IncrementalMiniBatchKMeansSampler(10, random_state=0)
+
+idx = ActiveLearningSplitter(10000, test_size=.2, random_state=0)
+idx.add_batch(np.arange(10))
+
+for i in range(10):
+
+    selected = sampler.fit(X[idx.selected]).select_samples(X[idx.non_selected], fixed_cluster_centers=X[idx.selected])
+
+    idx.add_batch(selected)
+    ikmeans_inertia.append(inertia(X[idx.test], X[idx.selected]))
+
+    clf.fit(X[idx.selected], y[idx.selected])
+    ikmeans_accuracy.append(clf.score(X[idx.test], y[idx.test]))
+
+
+
+irkmeans_inertia = []
+irkmeans_accuracy = []
+sampler = IncrementalMiniBatchKMeansSampler(10, random_state=0)
+
+idx = ActiveLearningSplitter(10000, test_size=.2, random_state=0)
+idx.add_batch(np.arange(10))
+
+for i in range(10):
+
+    selected = sampler.fit(X[idx.selected]).select_samples(X[idx.non_selected], fixed_cluster_centers=X[idx.selected], recenter_every=100)
+
+    idx.add_batch(selected)
+    irkmeans_inertia.append(inertia(X[idx.test], X[idx.selected]))
+
+    clf.fit(X[idx.selected], y[idx.selected])
+    irkmeans_accuracy.append(clf.score(X[idx.test], y[idx.test]))
```

Comment:
For the sake of this example as an intro to IWKMeans, perhaps we don't need to have the version with `recenter_every` ?

Reply:
As said, I will remove it!